src/core/vdom/patch.js

```diff
@@ -176,24 +177,15 @@ export function createPatchFunction (backend) {
       setScope(vnode)
 
       /* istanbul ignore if */
-      if (__WEEX__) {
+      if (__WEEX__ && (isUndef(data) || isFalse(data.appendAsTree))) {
```

Comment:
`data.appendAsTree` will be undefined when `append="tree"` is not present. Here it's better to just check `!data.appendAsTree`.